### ISSUE TITLE
[BUG] 除零风险 - prompt_length=0 时会导致失败

### ISSUE THREAD

**Issue by huthvincent** [huthvincent/KV-Cache-Compression-in-Reinforcement-Learning#3]
**Labels**: `bug`, `critical`, `edge-case`

### 问题描述
`SMD/src/shadow_mask_interceptor.py` 中 `_select_by_position_heuristic` 方法在 `prompt_length=0` 时会导致后续 `topk` 操作失败。

### 当前代码
```python
# 文件: SMD/src/shadow_mask_interceptor.py
# 行号: 约 200-210 行

def _select_by_position_heuristic(self, prompt_length: int, num_keep: int) -> torch.Tensor:
    # 缺少 prompt_length=0 的检查
    importance = torch.zeros(prompt_length, dtype=torch.float32)
    for i in range(prompt_length):
        # ...
    _, top_indices = importance.topk(num_keep)  # 如果 prompt_length=0，这里会失败
```

### 建议修复
```python
def _select_by_position_heuristic(self, prompt_length: int, num_keep: int) -> torch.Tensor:
    if prompt_length == 0:
        return torch.tensor([], dtype=torch.long)
    if num_keep >= prompt_length:
        return torch.arange(prompt_length)
    
    # ... 原有逻辑
```

### 影响范围
- 边界条件处理
- 可能在特殊输入时导致程序崩溃
